cluster/src/main/java/com/hartwig/pipeline/tertiary/lilac/Lilac.java

```diff
@@ -0,0 +1,105 @@
+    private final InputDownload purpleSomaticVcf;
+    private final InputDownload purpleSomaticVcfIndex;
+
+    public Lilac(final AlignmentPair alignmentPair, final ResourceFiles resourceFiles, final PurpleOutput purpleOutput) {
+        super(alignmentPair);
+        this.resourceFiles = resourceFiles;
+        this.purpleGeneCopyNumberTsv = new InputDownload(purpleOutput.outputLocations().geneCopyNumberTsv());
+        this.purpleSomaticVcf = new InputDownload(purpleOutput.outputLocations().somaticVcf());
+        this.purpleSomaticVcfIndex = new InputDownload(purpleOutput.outputLocations().somaticVcf().transform(x -> x + ".tbi"));
+    }
+
+    @Override
+    public String namespace() {
+        return NAMESPACE;
+    }
+
+    @Override
+    public List<BashCommand> inputs() {
+        List<BashCommand> result = new ArrayList<>(super.inputs());
+        result.add(purpleGeneCopyNumberTsv);
+        result.add(purpleSomaticVcf);
+        result.add(purpleSomaticVcfIndex);
+        return result;
+    }
+
+    @Override
+    public List<BashCommand> commands(final SomaticRunMetadata metadata) {
+        String slicedRefBam = VmDirectories.outputFile(metadata.reference().sampleName() + ".hla.bam");
+        String slicedTumorBam = VmDirectories.outputFile(metadata.tumor().sampleName() + ".hla.bam");
+        return List.of(new LilacBamSliceCommand(resourceFiles, getReferenceBamDownload().getLocalTargetPath(), slicedRefBam),
+                new LilacBamIndexCommand(slicedRefBam),
+                new LilacBamSliceCommand(resourceFiles, getTumorBamDownload().getLocalTargetPath(), slicedTumorBam),
+                new LilacBamIndexCommand(slicedTumorBam),
+                new LilacCommand(resourceFiles,
+                        metadata.tumor().sampleName(),
+                        slicedRefBam,
+                        slicedTumorBam,
+                        purpleGeneCopyNumberTsv.getLocalTargetPath(),
+                        purpleSomaticVcf.getLocalTargetPath()));
+    }
+
+    @Override
+    public VirtualMachineJobDefinition vmDefinition(final BashStartupScript bash, final ResultsDirectory resultsDirectory) {
+        return ImmutableVirtualMachineJobDefinition.builder()
+                .name("lilac")
+                .startupCommand(bash)
+                .performanceProfile(VirtualMachinePerformanceProfile.custom(8, 16))
+                .namespacedResults(resultsDirectory)
+                .build();
+    }
+
+    @Override
+    public LilacOutput output(final SomaticRunMetadata metadata, final PipelineStatus jobStatus, final RuntimeBucket bucket,
+            final ResultsDirectory resultsDirectory) {
+        return LilacOutput.builder()
+                .status(jobStatus)
+                .addFailedLogLocations(GoogleStorageLocation.of(bucket.name(), RunLogComponent.LOG_FILE))
+                .addReportComponents(new EntireOutputComponent(bucket, Folder.root(), namespace(), resultsDirectory))
```

Comment:
We'll want to add the Lilac loader to the DB loaders so we should add a datatype.